android/java/org/chromium/chrome/browser/brave_news/BraveNewsUtils.java

```diff
@@ -0,0 +1,128 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.brave_news;
+
+import android.content.Context;
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import org.chromium.base.Log;
+import org.chromium.brave_news.mojom.Article;
+import org.chromium.brave_news.mojom.CardType;
+import org.chromium.brave_news.mojom.Deal;
+import org.chromium.brave_news.mojom.FeedItem;
+import org.chromium.brave_news.mojom.DisplayAd;
+import org.chromium.brave_news.mojom.FeedItemMetadata;
+import org.chromium.brave_news.mojom.FeedPage;
+import org.chromium.brave_news.mojom.FeedPageItem;
+import org.chromium.brave_news.mojom.Image;
+import org.chromium.brave_news.mojom.PromotedArticle;
+import org.chromium.brave_news.mojom.Publisher;
+import org.chromium.chrome.browser.brave_news.models.FeedItemCard;
+import org.chromium.chrome.browser.brave_news.models.FeedItemsCard;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class BraveNewsUtils {
+    private Context mContext;
+
+    public BraveNewsUtils(Context context) {
+        mContext = context;
+    }
+
+    private Publisher getPublisher(){
+        Publisher publisher = null;
+
+        return publisher;
+
+    }
+
+    public String getPromotionIdItem(FeedItemsCard items) {
+        String creativeInstanceId = "null";
+        if (items.getFeedItems() != null){            
+            for (FeedItemCard itemCard : items.getFeedItems()) {
+                FeedItem item = itemCard.getFeedItem();
+                FeedItemMetadata itemMetaData = new FeedItemMetadata();
+                switch (item.which()) {
+                    case FeedItem.Tag.PromotedArticle:
+                        PromotedArticle promotedArticle = item.getPromotedArticle();
+                        FeedItemMetadata promotedArticleData = promotedArticle.data;
```

Comment:
`promotedArticleData` is not used, but below there is still `itemMetaData = promotedArticle.data;`

Reply:
this is still not fixed

Reply:
fixed now. the rest of the variables are just there to be used for logging. might remove the method entirely once we're comfortable with QA